### PR TITLE
docs: ui.button_group

### LINKED ISSUE #836: docs: ui.button_group

### PR DESCRIPTION
**Pull request by AkshatJawne**
Closes #836 

Did not include alignment section, there appears to be an issue with the `alignment` prop, more details included in this ticket: https://github.com/deephaven/deephaven-plugins/issues/909

### CODE REVIEW

File: plugins/ui/docs/components/button_group.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+## UI Recommendations
+
+Recommendations for creating button groups:
+
+1. The most critical action in a button group should use an accent, primary, or negative button style, while other actions should be secondary outline buttons.
+2. Button groups should be left-aligned to follow content such as blocks of text, center-aligned in empty states, and right-aligned in container components like dialogs, popovers, or cards.
+3. Button priority should match text alignment: for left-aligned text, the most critical button is on the left; for right- or center-aligned text, the most critical button is on the right.
+4. Icons should be used for higher priority actions if used in the button group. If the most critical action does not have an icon, avoid using icons for the other lower priority actions.
+
+Consider using a [`action_group`](./action_group.md) to allow the user to select from a list of actions.
```
Comment (margaretkennedy) [inline plugins/ui/docs/components/button_group.md:29]:
```suggestion
Consider using an [`action_group`](./action_group.md) to allow the user to select from a list of actions. 
```

File: plugins/ui/docs/components/button_group.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+2. Button groups should be left-aligned to follow content such as blocks of text, center-aligned in empty states, and right-aligned in container components like dialogs, popovers, or cards.
+3. Button priority should match text alignment: for left-aligned text, the most critical button is on the left; for right- or center-aligned text, the most critical button is on the right.
+4. Icons should be used for higher priority actions if used in the button group. If the most critical action does not have an icon, avoid using icons for the other lower priority actions.
+
+Consider using a [`action_group`](./action_group.md) to allow the user to select from a list of actions. 
+
+
+## Content
+
+Button group is used used to handle button overflow, and thus, expects buttons as children. It switches to a vertical layout when horizontal space is limited.
```
Comment (margaretkennedy) [inline plugins/ui/docs/components/button_group.md:34]:
```suggestion
A Button group is used to handle button overflow and, thus, expects buttons as children. It switches to a vertical layout when horizontal space is limited.
```

File: plugins/ui/docs/components/button_group.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
+# Button Group
+
+A button group is a UI component that groups buttons with related actions together.
+
+Note that the button component can only be used within a button group.
```
Comment (margaretkennedy) [inline plugins/ui/docs/components/button_group.md:5]: worth formatting as a true note?

> [!NOTE]
> The button component can only be used with a button group.

Comment (dsmmcken) [inline plugins/ui/docs/components/button_group.md:5]:
```suggestion
A button group is a UI component that groups buttons with related actions together and will automatically handle layout overflow nicely. Only buttons can be used within button group.
```

File: plugins/ui/docs/components/button_group.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+```python
+from deephaven import ui
+
+
+my_button_group_basic = ui.button_group(
+    ui.button("Rate Now", variant="primary"),
+    ui.button("No, thanks", variant="secondary"),
+    ui.button("Remind me later", variant="secondary"),
+)
```
Comment (dsmmcken) [inline plugins/ui/docs/components/button_group.md:17]:
```suggestion
from deephaven import ui


my_button_group_basic = ui.button_group(
    ui.button("Rate Now", variant="accent"),
    ui.button("No, thanks", variant="primary", style="outline"),
    ui.button("Remind me later", variant="primary", style="outline"),
)
```

Adjust to preferred styling.

File: plugins/ui/docs/components/button_group.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+```
+
+## UI Recommendations
+
+Recommendations for creating button groups:
+
+1. The most critical action in a button group should use an accent, primary, or negative button style, while other actions should be secondary outline buttons.
```
Comment (dsmmcken) [inline plugins/ui/docs/components/button_group.md:24]:
```suggestion
1. The most critical action in a button group should use an accent, or negative button style, while other actions should be primary outline buttons.
```

File: plugins/ui/docs/components/button_group.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    ui.button("Rate Now", variant="primary"),
+)
+
+
+my_button_group_content_with_less_space_example = ui.button_group(
+    ui.button("No, thanks", variant="secondary"),
+    ui.button("Remind me later", variant="secondary"),
+    ui.button("Rate Now", variant="primary"),
+    width=20,
+)
```
Comment (dsmmcken) [inline plugins/ui/docs/components/button_group.md:52]:
```suggestion
from deephaven import ui


my_button_group_content_space_example = ui.view(
    ui.button_group(
        ui.button("Rate Now", variant="accent", style="outline"),
        ui.button("Remind me later", variant="primary", style="outline"),
        ui.button("No, thanks", variant="primary", style="outline"),
    ),
    width=200,
    border_width="thin",
    padding="size-100",
)
```

It's more about the parent being constrained in width, and the group auto switching orientation. This shows that better.

File: plugins/ui/docs/components/button_group.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+The button group will remain in the orientation regardless of the width.
+
+```python
+from deephaven import ui
+
+
+my_button_group_orientation_example = ui.button_group(
+    ui.button("No, thanks", variant="secondary"),
+    ui.button("Remind me later", variant="secondary"),
+    ui.button("Rate Now", variant="primary"),
```
Comment (dsmmcken) [inline plugins/ui/docs/components/button_group.md:69]:
```suggestion
    ui.button("No, thanks", variant="primary", style="outline"),
    ui.button("Remind me later", variant="primary", style="outline"),
    ui.button("Rate Now", variant="accent"),
```

**Review by dsmmcken — CHANGES_REQUESTED** [review]

File: plugins/ui/docs/components/button_group.md (hunk 3 above)
Comment (dsmmcken) [reply plugins/ui/docs/components/button_group.md:5]: Changed suggested wording to note be a note.

**Review by AkshatJawne — COMMENTED** [review]
Ready for re-review @margaretkennedy @dsmmcken

**Review by dsmmcken — APPROVED** [review]

File: plugins/ui/docs/components/button_group.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
+# Button Group
+
+A button group is a UI component that groups buttons with related actions together and will automatically handle layout overflow nicely. Only buttons can be used within button group.
```
Comment (margaretkennedy) [inline plugins/ui/docs/components/button_group.md:3]:
```suggestion
A button group is a UI component that groups buttons with related actions together and will automatically handle layout overflow nicely. Only buttons can be used within button groups.
```

**Review by margaretkennedy — COMMENTED** [review]
One minor change.

**Review by dsmmcken — APPROVED** [review]

File: plugins/ui/docs/components/button_group.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+@ui.component
+def ui_button_group_alignment_examples():
+    return [
+        ui.button_group(
+            ui.button("No, thanks", variant="secondary"),
+            ui.button("Remind me later", variant="secondary"),
+            ui.button("Rate Now", variant="primary"),
+            orientation="vertical",
+        ),
```
Comment (dsmmcken) [inline plugins/ui/docs/components/button_group.md:86]: Remove default align, don't need to show this again.

File: plugins/ui/docs/components/button_group.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+            orientation="vertical",
+            align="center",
+        ),
+        ui.button_group(
+            ui.button("No, thanks", variant="secondary"),
+            ui.button("Remind me later", variant="secondary"),
+            ui.button("Rate Now", variant="primary"),
+            orientation="vertical",
+            align="end",
+        ),
```
Comment (dsmmcken) [inline plugins/ui/docs/components/button_group.md:100]: Use the variant + style combination as used in all the other examples pls.

**Review by dsmmcken — CHANGES_REQUESTED** [review]

**Review by dsmmcken — DISMISSED** [review]

File: plugins/ui/docs/components/button_group.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+```
+
+## UI Recommendations
+
+Recommendations for creating button groups:
+
+1. The most critical action in a button group should use an accent, or negative button style, while other actions should be primary outline buttons.
+2. Button groups should be left-aligned to follow content such as blocks of text, center-aligned in empty states, and right-aligned in container components like dialogs, popovers, or cards.
+3. Button priority should match text alignment: for left-aligned text, the most critical button is on the left; for right- or center-aligned text, the most critical button is on the right.
+4. Icons should be used for higher priority actions if used in the button group. If the most critical action does not have an icon, avoid using icons for the other lower priority actions.
```
Comment (margaretkennedy) [inline plugins/ui/docs/components/button_group.md:25]:
```suggestion
4. Icons should be used for higher-priority actions if used in the button group. If the most critical action does not have an icon, avoid using icons for the other lower-priority actions.
```

File: plugins/ui/docs/components/button_group.md [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+    ui.button("Remind me later", variant="primary", style="outline"),
+    ui.button("Rate Now", variant="accent"),
+    orientation="vertical",
+)
+```
+
+
+## Alignment
+
+By default, button groups are start-aligned to be in accordance with content, but they can be set to have a different alignment using the `alignment` prop.
```
Comment (margaretkennedy) [inline plugins/ui/docs/components/button_group.md:72]:
```suggestion
By default, button groups are start-aligned to accord with content, but they can be set to have a different alignment using the `alignment` prop.
```

**Review by dsmmcken — APPROVED** [review]

**Pull request merged**